### PR TITLE
First draft of my post Semi-Automatic App Versioning in Xamarin.Forms

### PR DESCRIPTION
**Pull request by spheroid-**

### CODE REVIEW

File: _posts/2020-05-12-semi-automatic-app-versioning-xamarin-forms.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+2. The actual internal version that the system uses to determine whether _version A_ is greater than _version B_. This is not visible to the user so the format here does not matter much – iOS uses a similarly formatted [CFBundleVersion](https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleversion) as above and on Android it's a single integer [versionCode](https://developer.android.com/studio/publish/versioning). We can, however, use the same format on both platforms as the iOS version can be abbreviated to a single integer. We'll call this one a **build number**.
+
+These are the versions that actually matter in the grand scheme of things, but a Xamarin.Forms project has even more versions (assembly versions, project versions and the solution version to name a few). What should we do with them?
+
+### My Prequisites
```
Comment (mattiuusitalo) [inline _posts/2020-05-12-semi-automatic-app-versioning-xamarin-forms.md:23]: I'm sure it's spelled _prerequisites_

File: _posts/2020-05-12-semi-automatic-app-versioning-xamarin-forms.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+* I want to be able to set the current app version in just one place.
+* I don't want to touch it until I've released the previous version and it's time to work on a new one.
+* Every build must be automatically numbered according to the Git revision (in a sensible way).
+* And there needs to be a way to work in a release branch that does not mess up the [trunk based development](https://trunkbaseddevelopment.com).
+
+## The Version Number
+
+There's a logical place where to set the canonical app version in Visual Studio – the **Solution Options**. On the **Main Settings** tab (it should be the default one when you open the settings window), you can enter the **Version** number. If you haven't modified the Android and iOS project configurations, they should inherit the version number from this value[^1].
```
Comment (mattiuusitalo) [inline _posts/2020-05-12-semi-automatic-app-versioning-xamarin-forms.md:32]: "they should inherit the version number from this value[^1]."

I don't understand this. What does it mean?

Comment (mattiuusitalo) [reply _posts/2020-05-12-semi-automatic-app-versioning-xamarin-forms.md:32]: Ah it's a reference. Maybe it renders well in the blog. I'm reading the diff.

File: _posts/2020-05-12-semi-automatic-app-versioning-xamarin-forms.md (hunk 1 above)
Comment (spheroid-) [reply _posts/2020-05-12-semi-automatic-app-versioning-xamarin-forms.md]: It is. I ran the entire post through spell check and corrected a couple of more errors as well.

File: _posts/2020-05-12-semi-automatic-app-versioning-xamarin-forms.md (hunk 2 above)
Comment (spheroid-) [reply _posts/2020-05-12-semi-automatic-app-versioning-xamarin-forms.md:33]: Yeah, they're footnotes :-)

**Pull request merged**